tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2681,35 +2683,50 @@ private String handleCurrencyFormat(XPathParts parts, String value, boolean show
 
         String numberSystem = parts.getAttributeValue(2, "numberSystem"); // null if not present
 
-        DecimalFormat df =
-                icuServiceBuilder.getCurrencyFormat(currency, currencySymbol, numberSystem);
-        df.applyPattern(value);
-
         String countValue = parts.getAttributeValue(-1, "count");
         if (countValue != null) {
+            DecimalFormat df =
+                icuServiceBuilder.getCurrencyFormat(currency, currencySymbol, numberSystem);
+            df.applyPattern(value);
             return formatCountDecimal(df, countValue);
         }
 
-        double sampleAmount = 1295.00;
-        example = addExampleResult(formatNumber(df, sampleAmount), example, showContexts);
-        example = addExampleResult(formatNumber(df, -sampleAmount), example, showContexts);
-
-        if (showContexts && !altAlpha) {
-            // If this example is not for alt="alphaNextToNumber", then if the currency symbol
-            // above has letters (strong dir) add another example with non-letter symbol
-            // (weak or neutral), or vice versa
-            if (symbolIsLetters(currencySymbol, false)) {
-                currency = "EUR";
-                checkPath = "//ldml/numbers/currencies/currency[@type=\"" + currency + "\"]/symbol";
-                currencySymbol = cldrFile.getWinningValue(checkPath);
-            } else {
-                currencySymbol = currency;
-            }
-            df = icuServiceBuilder.getCurrencyFormat(currency, currencySymbol, numberSystem);
-            df.applyPattern(value);
-            example = addExampleResult(formatNumber(df, sampleAmount), example, showContexts);
-            example = addExampleResult(formatNumber(df, -sampleAmount), example, showContexts);
+        String symbol_eur;
+        String symbol_usd;
+        boolean symbol_first = value.charAt(0) == '¤';
+        if (altAlpha) {
+            symbol_eur = "EUR";
+            symbol_usd = symbol_first ? "$USD" : "USD$";
+        }
+        else {
+            symbol_eur = "€";
+            symbol_usd = symbol_first ? "USD$" : "$USD";
         }
+        
+        String pattern_eur = value.replace("¤", symbol_eur);
+        String pattern_usd = value.replace("¤", symbol_usd);
+       
+        DecimalFormat df_eur = icuServiceBuilder.getNumberFormat(pattern_eur, numberSystem);
+        DecimalFormat df_usd = icuServiceBuilder.getNumberFormat(pattern_usd, numberSystem);
+
+        double sampleAmount = 1295.00;
+        example = addExampleResult(formatNumber(df_eur, sampleAmount), example, showContexts);
+        example = addExampleResult(formatNumber(df_eur, -sampleAmount), example, showContexts);
+        example = addExampleResult(formatNumber(df_usd, sampleAmount), example, showContexts);
+        example = addExampleResult(formatNumber(df_usd, -sampleAmount), example, showContexts);
+
+        DecimalFormat df = icuServiceBuilder.getCurrencyFormat(currency, "¤", numberSystem);
```

Comment:
Passing "¤" is ugly but seems to work since it is not one of the characters that triggers space insertion